### ISSUE TITLE
How compact should we make the JSON and YAML representations of SPDX documents?

### ISSUE THREAD

**Issue by goneall** [spdx/spdx-spec#189]
In writing tools to support JSON and YAML, I ran into an issue on how to represent property values of an SPDX element type.

One approach would be to always include the full exploded SPDX element.  This can create a huge document with a LOT of redundancy since the same element may be referenced by several different properties.

Another approach would be to use the SPDX ID and store one copy of the SPDX element in some other portion of the document (e.g. in an SPDX element array proposed in issue #188 )

A similar issue exists for licenses.  We could include license references with all elements of the license included in the document (e.g. an SPDX listed license would include the name, text, etc.).

**Comment by goneall** [comment]
For the SPDX tool, I created an option for the JSON format:

1) COMPACT - only use SPDX ID's and license expressions
2) STANDARD - fully represent every reference to an SPDX element but use license expressions
3) FULL - fully represent license expressions and SPDX elements

I've attached an archive file with the SPDX examples in these 3 different formats.
[JsonExamples.zip](https://github.com/spdx/spdx-spec/files/4112947/JsonExamples.zip)

**Comment by tsteenbe** [comment]
In the SPDX tech call Jan 28, 2020 we decided to go with compact version.

```      
  "hasFile": [
        "SPDXRef-JenaLib",
        "SPDXRef-DoapSource",
         "SPDXRef-CommonsLangSrc"
    ],
```

with: @goneall  @swinslow  @nishakm @rnjudge, Rohit and Jim Hutchison

**Comment by goneall** [comment]
Resolved with PR #197
